cpp/src/io/csv/csv_gpu.cu

```diff
@@ -1014,23 +1016,29 @@ void __host__ remove_blank_rows(rmm::device_vector<uint64_t> &row_offsets,
   row_offsets.resize(new_end - row_offsets.begin());
 }
 
-cudaError_t __host__ DetectColumnTypes(const char *data,
-                                       const uint64_t *row_starts,
-                                       size_t num_rows,
-                                       size_t num_columns,
-                                       const ParseOptions &options,
-                                       column_parse::flags *flags,
-                                       column_parse::stats *stats,
-                                       cudaStream_t stream)
+std::vector<column_parse::stats> detect_column_types(const char *data,
+                                                     const uint64_t *row_starts,
+                                                     size_t num_rows,
+                                                     size_t num_actual_columns,
+                                                     size_t num_active_columns,
+                                                     const cudf::io::ParseOptions &options,
+                                                     column_parse::flags *flags,
+                                                     cudaStream_t stream)
 {
   // Calculate actual block count to use based on records count
   const int block_size = csvparse_block_dim;
   const int grid_size  = (num_rows + block_size - 1) / block_size;
 
+  auto d_stats = rmm::device_vector<column_parse::stats>(num_active_columns);
+
   data_type_detection<<<grid_size, block_size, 0, stream>>>(
-    data, options, num_rows, num_columns, flags, row_starts, stats);
+    data, options, num_rows, num_actual_columns, flags, row_starts, d_stats.data().get());
 
-  return cudaSuccess;
+  auto h_stats = std::vector<column_parse::stats>(num_active_columns);
+
+  thrust::copy(d_stats.begin(), d_stats.end(), h_stats.begin());
+
+  return h_stats;
```

Comment:
```suggestion
  return std::vector<column_parse::stats>(d_stats.begin(), d_stats.end());
```

Reply:
I don't think this works since `d_stats` is a device vector.

Reply:
Make `h_stats` a `thrust::host_vector`.